src/scenechange/mod.rs

```diff
@@ -194,51 +170,57 @@ impl<T: Pixel> SceneChangeDetector<T> {
         frame_set[self.deque_offset].clone(),
         frame_set[self.deque_offset + 1].clone(),
         input_frameno,
-        previous_keyframe,
       );
     } else {
       self.deque_offset -= 1;
     }
 
     // Adaptive scenecut check
-    let scenecut = self.adaptive_scenecut();
+    let (mut scenecut, score) = self.adaptive_scenecut();
+    if let Some(result) = self.handle_min_max_intervals(distance) {
```

Comment:
let scenecut = self.handle_min_max_intervals(distance).unwrap_or(scenecut);

maybe?

Reply:
That does seem a little cleaner, yeah.